build/package/nfpm.yaml

```diff
@@ -38,6 +38,10 @@ contents:
   dst: /lib/systemd/system/kong.service
 - src: build/package/kong.logrotate
   dst: /etc/kong/kong.logrotate
+- dst: /var/run/kong
```

Comment:
`/var/run` is mostly a symlink to `/run`, which is a volatile tmpfs. So this directory shouldn't be created by package installer. Instead, use something like `ReadWritePaths=-/var/run/kong` in systemd service file, and `mkdir -p` in docker entrypoint.

Reply:
> `/var/run` is mostly a symlink to `/run`, which is a volatile tmpfs. So this directory shouldn't be created by package installer. Instead, use something like `ReadWritePaths=-/var/run/kong` in systemd service file, and `mkdir -p` in docker entrypoint.

I'm not sure how to resolve this -- the /var/run and /run dirs will require root/sudo to create subdirectories. When using package installer, the preferred method of running Kong would be systemd but I know from experience that users sometimes just use `kong start` instead. In docker, we don't run as root so would `mkdir -p` be able to create directories or files in /var/run?

Reply:
if the final location is passed along within Kong as a config item, can you then on start detect if the proper path is available and accessible, and if not fallback to the curent location (in the prefix), whilst logging a warning?

From there on all Kong code would use the location specified in the config item.

Reply:
I found a possible solution using tmpfiles.d and I've updated the nfpm yaml and the dockerfiles. There is no systemd in docker containers so there we need to create the directories. I updated the postinstall.sh to call `systemd-tmpfiles --create` in order to create the tmpfiles on install (otherwise they wouldn't be created until the system is rebooted, I believe).

Reply:
@brentos We will actually need both the systemd file modification and the docker solution.
The systemd-tmpfiles solution won't work if the system is rebooted when kong is running outside of container. Use `ReadWritePaths=-/var/run/kong` so that the directory is created each time before kong started.
And plus the Dockerfile approach you already had in this PR (for containers).